src/canisters/frontend/ic-certified-assets/src/state_machine.rs

```diff
@@ -459,6 +462,9 @@ impl State {
 
         if let Some(certificate_header) = index_redirect_certificate {
             if let Some(asset) = self.assets.get(INDEX_FILE) {
+                if asset.allow_raw_access.map_or(true, |v| !v) && req.is_raw_domain() {
```

Comment:
For clarity.  With `map_or`, default of true (the inverse of the default of `allow_raw_access`), and negating the value in the closure, there's a lot going on here.
```suggestion
                if !asset.allow_raw_access() && req.is_raw_domain() {
```

along with
```
impl Asset {
    pub fn allow_raw_access(&self) -> bool {
        self.allow_raw_access.unwrap_or(false)
    }
}

```